### ISSUE TITLE
Leaking file handles

### ISSUE THREAD

**Issue by cclauss** [DavidBuchanan314/dlinject#1]
File handles are being leaked (__open()__ with no __close()__) at:
* https://github.com/DavidBuchanan314/dlinject/blob/3a29243fe78edf56e97c460d248491cf76cc9db1/dlinject.py#L11
* https://github.com/DavidBuchanan314/dlinject/blob/3a29243fe78edf56e97c460d248491cf76cc9db1/dlinject.py#L27

**Comment by DavidBuchanan314** [comment]
I believe python closes the file object once it goes out of scope, which `strace` output would appear to confirm:

```
openat(AT_FDCWD, "/proc/42403/maps", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
ioctl(4, TCGETS, 0x7ffefca47ba0)        = -1 ENOTTY (Inappropriate ioctl for device)
lseek(4, 0, SEEK_CUR)                   = 0
ioctl(4, TCGETS, 0x7ffefca47ae0)        = -1 ENOTTY (Inappropriate ioctl for device)
read(4, "55bff1afd000-55bff1afe000 r--p 0"..., 8192) = 2985
read(4, "", 8192)                       = 0
close(4)                                = 0
...
openat(AT_FDCWD, "/proc/42403/stat", O_RDONLY|O_CLOEXEC) = 6
fstat(6, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
ioctl(6, TCGETS, 0x7ffc0784f1b0)        = -1 ENOTTY (Inappropriate ioctl for device)
lseek(6, 0, SEEK_CUR)                   = 0
ioctl(6, TCGETS, 0x7ffc0784f0f0)        = -1 ENOTTY (Inappropriate ioctl for device)
lseek(6, 0, SEEK_CUR)                   = 0
fstat(6, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
read(6, "42403 (test) T 42309 42403 42309"..., 8192) = 314
read(6, "", 7878)                       = 0
close(6)                                = 0
```

Admittedly, it might be better practice to use a `with` block.

**Comment by cclauss** [comment]
CPython does but that is not guaranteed by the Python Language Specification and is not supposed to be relied upon.
* https://stackoverflow.com/questions/7395542/is-explicitly-closing-files-important

**Comment by DavidBuchanan314** [comment]
I see, thank you. I believe I've fixed the issue in e5462717e5b2ebf99e46f7cd69a2b31ab68f7ad6
